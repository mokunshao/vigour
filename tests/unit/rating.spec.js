import { mount } from '@vue/test-utils';
import Rating from '../../src/components/rating.vue';

describe('Rating', () => {
  it('Rating 存在', () => {
    expect(Rating).toBeTruthy();
  });
  test('renders correctly', () => {
    const wrapper = mount(Rating, {
      propsData: {
        value: 3,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    wrapper.setProps({ value: 1 });
    expect(wrapper.element).toMatchSnapshot();
  });
});
