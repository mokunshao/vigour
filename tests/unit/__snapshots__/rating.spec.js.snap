// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders correctly 1`] = `
<div
  class="vigour-rating"
>
  <span
    class="vigour-rating-star vigour-rating-star-active"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star vigour-rating-star-active"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star vigour-rating-star-active"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star"
  >
    
    ★
  
  </span>
</div>
`;

exports[`Rating renders correctly 2`] = `
<div
  class="vigour-rating"
>
  <span
    class="vigour-rating-star vigour-rating-star-active"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star"
  >
    
    ★
  
  </span>
  <span
    class="vigour-rating-star"
  >
    
    ★
  
  </span>
</div>
`;
